frontend/src/services/UserFunctions.ts

```diff
@@ -1,15 +1,16 @@
 import { UserCredential, getAuth, onAuthStateChanged } from "firebase/auth"; // Import UserCredential type
 import { SuccessObject, callUserFunction } from "@/lib/utils";
 
-export async function addToUserCollection(userCredential: UserCredential): Promise<SuccessObject> {
+export async function addToUserCollection(userCredential: UserCredential, username: string): Promise<SuccessObject> {
     const { uid, email, displayName } = userCredential.user;
     // Prepare the data to send in the request body
     const userData = {
         uid,
         email,
         displayName: displayName || '', // Use an empty string if displayName is not provided
+        username: username,
     };
-    console.log("sending user to collection...")
+    console.log("sending user to collection..." + userData)
```

Comment:
imo if this is used for testing only, should remove before PR

Reply:
removed